ptf/saimpls.py

```diff
@@ -0,0 +1,3944 @@
+        '''
+        print("\nmplsEgressPhpTtlModeTest()")
+
+        send_pkt = simple_tcp_packet(
+            eth_dst=ROUTER_MAC,
+            ip_dst='30.30.30.1',
+            ip_ttl=64)
+
+        mpls_tag = {'label': 2000, 'ttl': 60, 'tc':  0, 's':  1}
+        mpls_tag_list = [mpls_tag]
+        mpls_pkt = simple_mpls_packet(
+            eth_dst=ROUTER_MAC,
+            mpls_tags=mpls_tag_list,
+            inner_frame=send_pkt['IP'])
+
+        mpls_tag_2 = {'label': 2111, 'ttl': 58, 'tc':  0, 's':  0}
+        mpls_tag_list_2 = [mpls_tag_2, mpls_tag]
+        mpls_tag_2_pkt = simple_mpls_packet(
+            eth_dst=ROUTER_MAC,
+            mpls_tags=mpls_tag_list_2,
+            inner_frame=send_pkt['IP'])
+
+        mpls_tag_3 = {'label': 2222, 'ttl': 56, 'tc':  0, 's':  0}
+        mpls_tag_list_3 = [mpls_tag_3, mpls_tag_2, mpls_tag]
+        mpls_tag_3_pkt = simple_mpls_packet(
+            eth_dst=ROUTER_MAC,
+            mpls_tags=mpls_tag_list_3,
+            inner_frame=send_pkt['IP'])
+
+        recv_pkt = simple_tcp_packet(
+            eth_dst='00:11:22:33:44:55',
+            eth_src=ROUTER_MAC,
+            ip_dst='30.30.30.1',
+            ip_ttl=63)
+        
+        mpls_tag['ttl'] = 55
+        recv_mpls_tag_list = [mpls_tag]
+        recv_mpls_pkt = simple_mpls_packet(
+            eth_dst='00:11:22:33:44:55',
+            eth_src=ROUTER_MAC,
+            mpls_tags=recv_mpls_tag_list,
+            inner_frame=send_pkt['IP'])
+
+        try:
+            print("Send MPLS tag pakcet with label 2000 - "
+                  "PHP and forward IP packet")
+            recv_pkt['IP'].ttl = 59
+            send_packet(self, self.dev_port10, mpls_pkt)
+            verify_packet(self, recv_pkt, self.dev_port13)
+
+            print("Send MPLS tag pakcet with labels - 2111, 2000 - "
+                  "PHP and forward IP packet")
+            recv_pkt['IP'].ttl = 57
+            send_packet(self, self.dev_port10, mpls_tag_2_pkt)
+            verify_packet(self, recv_pkt, self.dev_port13)
+
+            print("Send MPLS tag pakcet with labels - 2222, 2111, 2000 - "
+                  "PHP and forward IP packet")
+            recv_pkt['IP'].ttl = 55
+            send_packet(self, self.dev_port10, mpls_tag_3_pkt)
+            verify_packet(self, recv_pkt, self.dev_port13)
+
+            print("Set MPLS object label 2222 to pop only 2 labels")
+            sai_thrift_set_inseg_entry_attribute(
+                self.client, self.inseg_entry_2222, num_of_pop=2)
+            print("Send MPLS tag pakcet with labels - 2222, 2111, 2000 - "
+                  "Pop 2 labels and forward MPLS with 2000")
+            send_packet(self, self.dev_port10, mpls_tag_3_pkt)
+            verify_packet(self, recv_mpls_pkt, self.dev_port13)
+
+        finally:
+            sai_thrift_set_inseg_entry_attribute(
+                self.client, self.inseg_entry_2222, num_of_pop=3)
+
+    def mplsTransitPushTtlModeTest(self):
```

Comment:
I no longer believe that the TTL values in this test are correct from a standards point of view.
This test should be deleted for now, can be restored once we put in a fix.

I opened SWI-4749 to track this.